core/lib/dal/src/storage_logs_dedup_dal.rs

```diff
@@ -172,28 +172,45 @@ impl StorageLogsDedupDal<'_, '_> {
         .map(|max| max as u64))
     }
 
-    /// Returns the maximum enumeration index assigned in a specific L1 batch.
-    pub async fn max_enumeration_index_for_l1_batch(
+    pub async fn max_enumeration_index_in_l1_batch(
         &mut self,
         l1_batch_number: L1BatchNumber,
     ) -> DalResult<Option<u64>> {
-        let row = sqlx::query!(
+        let Some(max_l1_batch_number) = sqlx::query!(
```

Comment:
Dumb question: Can't this be a single query, e.g. using a subquery? AFAIU, using a "plain" query won't do because it would involve a full `initial_writes` scan at the worst case.

Reply:
Yes, all correct, I will refactor to use subquery